### PR TITLE
vk: Add support for VK_EXT_attachment_feedback_loop_layout

### PR DESCRIPTION
**Pull request by kd-11**
Allows us to choose to disable DCC/HTILE when we know a render target surface is being used for both read and write simultaneously. This fixes the diagonal artifact seen when using rpcs3 on some platforms. We have some workarounds in place already, but they negatively affect performance.

This extension is currently only present in RADV.

### CODE REVIEW

File: rpcs3/Emu/RSX/VK/vkutils/device.h [hunk 1]
```diff
@@ -132,37 +133,38 @@ namespace vk
...
+		bool get_framebuffer_loops_support() const { return pgpu->framebuffer_loops_support; }
+
+		u64 get_descriptor_update_after_bind_support() const { return pgpu->descriptor_update_after_bind_mask; }
+		u32 get_descriptor_max_draw_calls() const { return pgpu->descriptor_max_draw_calls; }
+
+		VkQueue get_present_queue() const { return m_present_queue; }
+		VkQueue get_graphics_queue() const { return m_graphics_queue; }
+		VkQueue get_transfer_queue() const { return m_transfer_queue; }
+		u32 get_graphics_queue_family() const { return m_graphics_queue_family; }
+		u32 get_present_queue_family() const { return m_graphics_queue_family; }
```
Comment (Megamouse) [inline rpcs3/Emu/RSX/VK/vkutils/device.h:162]: typo?

File: rpcs3/Emu/RSX/VK/VulkanAPI.h [hunk 2]
```diff
@@ -21,3 +21,18 @@
...
+#define VK_EXT_attachment_feedback_loop_layout 1
+#define VK_EXT_ATTACHMENT_FEEDBACK_LOOP_LAYOUT_EXTENSION_NAME "VK_EXT_attachment_feedback_loop_layout"
+#define VK_IMAGE_LAYOUT_ATTACHMENT_FEEDBACK_LOOP_OPTIMAL_EXT static_cast<VkImageLayout>(1000339000)
+#define VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_ATTACHMENT_FEEDBACK_LOOP_LAYOUT_FEATURES_EXT static_cast<VkStructureType>(1000339000)
+
+typedef struct VkPhysicalDeviceAttachmentFeedbackLoopLayoutFeaturesEXT {
+	VkStructureType    sType;
+	void* pNext;
+	VkBool32           attachmentFeedbackLoopLayout;
+} VkPhysicalDeviceAttachmentFeedbackLoopLayoutFeaturesEXT;
```
Comment (Megamouse) [inline rpcs3/Emu/RSX/VK/VulkanAPI.h:36]: format?

File: rpcs3/Emu/RSX/VK/vkutils/device.h (hunk 1 above)
Comment (kd-11) [reply rpcs3/Emu/RSX/VK/vkutils/device.h:162]: No, we don't support a standalone present queue at the moment. Present and graphics share a queue.

File: rpcs3/Emu/RSX/VK/VulkanAPI.h (hunk 2 above)
Comment (kd-11) [reply rpcs3/Emu/RSX/VK/VulkanAPI.h]: fixed

File: rpcs3/Emu/RSX/VK/VKDraw.cpp [hunk 3]
```diff
@@ -565,54 +571,7 @@ bool VKGSRender::bind_texture_env()
...
-				{
-					src_stage = VK_PIPELINE_STAGE_LATE_FRAGMENT_TESTS_BIT;
-					src_access = VK_ACCESS_DEPTH_STENCIL_ATTACHMENT_WRITE_BIT;
-				}
-
-				vk::insert_image_memory_barrier(
-					*m_current_command_buffer,
-					raw->value,
-					VK_IMAGE_LAYOUT_GENERAL, VK_IMAGE_LAYOUT_SHADER_READ_ONLY_OPTIMAL,
-					src_stage, VK_PIPELINE_STAGE_VERTEX_SHADER_BIT,
```
Comment (Megamouse) [inline rpcs3/Emu/RSX/VK/VKDraw.cpp:603]: VK_PIPELINE_STAGE_VERTEX_SHADER_BIT seems to differ here.

**Pull request merged**